### ISSUE TITLE
Re-add ruler-derived tests to CI

### ISSUE THREAD

**Issue by timbray** [timbray/quamina#176]
The two unit tests adapted from AWS Event Ruler, cl2_test.go and generic_machine_test.go, have been given the `fromRuler` build tag. Thus, they don't run in the CI task that runs unit tests. This was necessary because they stressed the GitHub task runner to the point that they the unit-test run was asynchronously interrupted with resulting code 143.

They are useful tests and should be included.  Suggestion: make another task that just runs these tests. If you say
`go test -run X`  
it runs all the tests whose func names match X, which is a regexp.

I've tested and 

```sh
go test -v -run 'TestArrays|TestNested|TestSimplestPossibleMachine|TestEmptyInput|TestCl2'
```

works. But I think it would be better to rename all those tests with func names. beginning TestRuler, then say
```sh
go test -run  '^TestRuler' 
```

I have verified that this would work.

**Comment by timbray** [comment]
This could be done as another job in the existing go-unit-tests workflow, or in a newly created workflow, which would look a lot like the existing go-unit-tests except for the details of the actual `go test` command.

**Comment by embano1** [comment]
Please file the build tag change as a PR. I'll do a follow up PR to fix the unit test w/out duplication.

**Comment by embano1** [comment]
Related question: do you want to also run unit tests on Windows?

**Comment by timbray** [comment]
> Related question: do you want to also run unit tests on Windows?

Wow, that never crossed my mind. It's hard to believe that this kind of bit-twiddling no-runtime-dependency code would have any OS-lined problems, no?  Or is there experience in the Go community that suggests it would be worthwhile?

**Comment by timbray** [comment]
> Please file the name and build tag (if needed) changes as a PR. I'll do a follow up PR to fix the unit test w/out duplication.

OK.
> 
> Note: we still need a build tag to easily exclude them from the default tests w/out having to fiddle with regex there too.

Why do we want to exclude them from the default tests?  I think we only want to exclude them from the main unit-test action and strictly to work around the stupid GitHub bug.

**Comment by timbray** [comment]
Fixed in 56a7b7c0a4f2dcf4623ce81771adf35145ad6a80
